src/benchmarks/micro/libraries/System.IO/StringReaderReadLineTests.cs

```diff
@@ -0,0 +1,118 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace System.IO.Tests
+{
+    [BenchmarkCategory(Categories.Libraries)]
+    public class StringReaderReadLineTests
+    {
+        // Benchmark is based on measuring individual `ReadLine` calls
+        // of lines of different lengths within a range defined as
+        // benchmark parameters.
+        // To fulfill BDNs requirement for an iteration time of 250ms,
+        // this means a lot of invocations, more than a single string in
+        // a `StringReader` can accommodate.
+        // Therefore, the benchmark generates of string with a certain
+        // target length and then a number of readers for the same
+        // string, since `StringReader` does not support being "reset".
+        // The trade off is we need to switch reader during the `ReadLine`
+        // call when current reader ends, which adds a bit of overhead. 
+        // This is deemed negligible and amortized constant.
+        private const int StringReaderTargetLength = 16 * 1024 * 1024;
+        // Reader count must be high enough to ensure we never run out
+        // of new readers during benchmarking. Otherwise, it will throw.
+        private const int ReaderCount = 10000;
+
+        private string _text;
+        private StringReader[] _readers;
+        private int _readerIndex;
+        private StringReader _reader;
+
+        public StringReaderReadLineTests()
+        {
+            LineLengthRanges = new Range[]
+            {
+                new (){ Min =   0, Max =    0 },
+                new (){ Min =   1, Max =    8 },
+                new (){ Min =   9, Max =   32 },
+                new (){ Min =  33, Max =  128 },
+                new (){ Min = 129, Max = 1024 },
+            };
+        }
```

Comment:
I'd expect it'd be fairly common to find real world data that alternates between longer line lengths and empty lines,  with paragraphs separated by blank lines.

Reply:
I should note that the test cases exactly match real world data and use case for me, namely `csv` content where lines are never empty and often have same range of lengths.

Reply:
one way to add this would be to formalize a `EmptyLineRatio` e.g. 0.20 as part of `Range` (that should then be renamed). Not sure there is much value in it overall, though. It's pretty straightforward, very short lines see regressions. Others, large gains. If you mix the two, the large gains will prevail since they dominate completely. E.g. A 1024 char line takes much longer to "find" than 1 char line. If that 1 char line only occurs 20% of the time. It's pretty irrelevant. :)

Reply:
Or it other words -200ns compared to +3ns every 5 lines is not really interesting :)

Reply:
Makes sense to me. @stephentoub ?

Reply:
I was primarily pointing out that it's not clear to me how these ranges were arrived at, e.g. why do we think a min of 33 and a max of 128 is interesting to encode in a perf test?  What is the randomness within these ranges buying us?  Is this trying to simulate typical inputs in some file format (csv, xml, json, whatever) that we believe is used with StringReader frequently and such file formats typically have such line lengths?